ui/raidboss/data/06-ew/raid/p8s.ts

```diff
@@ -2064,54 +2086,277 @@ const triggerSet: TriggerSet<Data> = {
       },
     },
     {
-      id: 'P8S Perfected Alpha',
+      id: 'P8S Perfection Splicer Collect',
+      // Record what the splicers chose as they will marge with unused short player
       type: 'GainsEffect',
-      netRegex: NetRegexes.gainsEffect({ effectId: 'D05' }),
-      condition: Conditions.targetIsYou(),
-      // TODO: it'd be nice to know the tower here so this could just say
-      // "take tower" or "avoid tower" with different severity or even
-      // who to merge with (!), but without that this is the best we got.
-      infoText: (_data, _matches, output) => output.text!(),
-      outputStrings: {
-        text: {
-          en: 'Green/Blue Tower',
-          de: 'Grüner/Blauer Turm',
-          fr: 'Tour Verte/Bleue',
-          ja: '緑・青',
-          ko: '초록/파랑 기둥',
-        },
+      netRegex: NetRegexes.gainsEffect({ effectId: ['D05', 'D06', 'D07'] }),
+      condition: (data, matches) => {
+        // Ignore Imperfection players, and do not collect on HC2
+        if (data.perfectionLong[matches.target] || data.perfectionShort[matches.target] || data.arcaneChannelCount > 1)
+          return false;
+        return true;
+      },
+      run: (data, matches) => {
+        let letter;
+        if (matches.effectId === 'D05')
+          letter = 'alpha';
+        else if (matches.effectId === 'D06')
+          letter = 'beta';
+        else
+          letter = 'gamma';
+
+        data.perfectionShort[matches.target] = letter;
       },
     },
     {
-      id: 'P8S Perfected Beta',
+      id: 'P8S Perfection Remove',
+      // Remove player from perfection lists when they merge
       type: 'GainsEffect',
-      netRegex: NetRegexes.gainsEffect({ effectId: 'D06' }),
-      condition: Conditions.targetIsYou(),
-      infoText: (_data, _matches, output) => output.text!(),
-      outputStrings: {
-        text: {
-          en: 'Green/Purple Tower',
-          de: 'Grüner/Lilaner Turm',
-          fr: 'Tour Verte/Violette',
-          ja: '緑・紫',
-          ko: '초록/보라 기둥',
-        },
+      netRegex: NetRegexes.gainsEffect({ effectId: ['D09', 'D0A', 'D0B', 'D0C'] }),
+      run: (data, matches) => {
+        delete data.perfectionLong[matches.target];
+        delete data.perfectionShort[matches.target];
       },
     },
     {
-      id: 'P8S Perfected Gamma',
-      type: 'GainsEffect',
-      netRegex: NetRegexes.gainsEffect({ effectId: 'D07' }),
-      condition: Conditions.targetIsYou(),
-      infoText: (_data, _matches, output) => output.text!(),
-      outputStrings: {
-        text: {
-          en: 'Purple/Blue Tower',
-          de: 'Lilaner/Blauer Turm',
-          fr: 'Tour Violette/Bleue',
-          ja: '紫・青',
-          ko: '보라/파랑 기둥',
-        },
+      id: 'P8S Arcane Channel Collect',
+      type: 'MapEffect',
+      netRegex: NetRegexes.mapEffect({ flags: arcaneChannelFlags }),
+      condition: (data) => data.seenFirstTankAutos,
```

Comment:
I believe the same mapEffect flags exist in phase 1, so I added the condition to prevent further checks on the location value that are in the run section.